scripts/lib/run-paths.sh

```
@@ -0,0 +1,57 @@
+# Resolve the worktree root. `git rev-parse` is preferred so cd'ing into a
+# subdirectory still yields the worktree root, but fall back to $PWD for the
+# rare case where this is sourced outside a git checkout.
+sw_worktree_root="$(git rev-parse --show-toplevel 2>/dev/null || pwd)"
+sw_worktree_basename="$(basename "$sw_worktree_root")"
+
+# Guard against an empty basename. `basename /` returns "/" which would make
+# the run dir `$SW_RUN_ROOT//` and silently merge artifacts from any "root"
+# invocation into the same bucket. Surface the misconfiguration loudly.
+if [ -z "$sw_worktree_basename" ] || [ "$sw_worktree_basename" = "/" ]; then
+  echo "scripts/lib/run-paths.sh: cannot derive worktree basename from '$sw_worktree_root'" >&2
+  return 1 2>/dev/null || exit 1
```

Shellcheck (0.11.0)

[info] Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

(SC2317)